.circleci/config.yml

```diff
@@ -0,0 +1,216 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: edxops/xenial-common:hawthorn.beta1
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+
+      - run:
+          name: Do base configuration
+          command: |
+            ./scripts/circle-ci-configuration.sh
+
+      - save_cache:
+          key: v1-{{ checksum ".circleci/config.yml" }}-configuration-{{ checksum "scripts/circle-ci-configuration.sh" }}
+          paths:
+            - "downloads"
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-{{ checksum "requirements/edx/pre.txt" }}-{{ checksum "requirements/edx/paver.txt" }}-{{ checksum "requirements/edx/github.txt" }}-{{ checksum "requirements/edx/base.txt" }}-{{ checksum "requirements/edx/local.txt" }}-{{ checksum "requirements/edx/post.txt" }}
+            - v1-{{ checksum ".circleci/config.yml" }}-pip-deps-
+
+      - run:
+          name: Install pip packages
+          command: |
+
+            source venv/bin/activate
+
+            # All files listed here must be included in the cache key for pip packages.
+            pip install --exists-action w -r requirements/edx/pre.txt
+            pip install --exists-action w -r requirements/edx/paver.txt
+            pip install --exists-action w -r requirements/edx/github.txt || EXIT=1
+            pip install --exists-action w -r requirements/edx/github.txt # recommender-xblock installation fails first time; so try again.
+            pip install --exists-action w -r requirements/edx/base.txt
+            pip install --exists-action w -r requirements/edx/local.txt
+            pip install --exists-action w -r requirements/edx/post.txt
+
+      - save_cache:
+          key: v1-{{ checksum ".circleci/config.yml" }}-pip-deps-{{ checksum "requirements/edx/pre.txt" }}-{{ checksum "requirements/edx/paver.txt" }}-{{ checksum "requirements/edx/github.txt" }}-{{ checksum "requirements/edx/base.txt" }}-{{ checksum "requirements/edx/local.txt" }}-{{ checksum "requirements/edx/post.txt" }}
+          paths:
+            - "venv"
+
+      - restore_cache:
+          keys:
+            - v1-{{ checksum ".circleci/config.yml" }}-npm-deps-{{ checksum "package.json" }}
+            - v1-{{ checksum ".circleci/config.yml" }}-npm-deps-
+      - run:
+          name: Install npm packages
+          command: |
+            npm install
+
+      - save_cache:
+          key: v1-{{ checksum ".circleci/config.yml" }}-npm-deps-{{ checksum "package.json" }}
+          paths:
+            - "node_modules"
+
+      - run:
+          name: Done
+          command: |
+            echo "---------- DONE ----------"
```

Comment:
nit: hm, this echo doesn't really help in any way, just adds another step on Circle and seems unuseful.

Reply:
I'll remove all these.